src/services/validation.js

```diff
@@ -4,23 +4,21 @@ import firebaseAPI from "@/services/API/checkUser.js";
 * firebaseAPI service is used to validate and it returns a boolean value, indicating whether the user is valid or not.
 * @param {String} userID - current ID being authenticated
 * @param {Number} validateCount - indicates how many times the user has been validated
+* @param {String} collectionName - firestore collection against which the ID needs to be validated
+* @param {String} columnName - name of the column which contains the ID
 */
 
-export async function validateSRN(userID, validateCount){
-    let invalidLoginMessage = ""
-
-    let isCurrentUserValid = await firebaseAPI.checkUserExists(userID);
+export async function validateID(userID, validateCount, collectionName, columnName){
```

Comment:
We'll have to think this through thoda. I think we were also supposed to support different **_types_** of datastores right?

In the JSON that you've written in the doc, I also see a `type` option. (attaching a screenshot)
<img width="300" alt="Screenshot 2021-12-16 at 7 41 02 PM" src="https://user-images.githubusercontent.com/20169829/146387501-18de37d5-c101-412f-994f-7e9833043e35.png">

But I don't see any code which is checking the type and then redirecting to different methods? Althought right now we only have one type but it's not being checked anywhere. 

Ideally this file, `validation.js` should deal with all kinds of validations from all kinds of data sources. The method called `validateID` should take in the `userID`, `validateCount` and **_all the datastore details_**. 

When we're just passing `collectionName` and `columnName`, we're coupling this validation method to a firebase type datastore. I'm guessing you would've thought of that already. We'll deal with other kinds of datastore when they come but we should start building in the support! Like, in our case currently, pass the whole datastore object, check it's type, and call the `firebaseAPI.checkUserExists()` method if the type is firebase.

We'll discuss more on the call!